terraform/aws/vpc/main.tf

```diff
@@ -1,27 +1,83 @@
-locals {}
-
-provider "aws" {
-  region = var.region
-}
-
+# If vpc_id = "create" use module to create new VPC
 module "vpc" {
-  source  = "terraform-aws-modules/vpc/aws"
-  version = "2.15.0"
+  create_vpc = (var.vpc_id == "create" ? "true" : "false")
+  source     = "terraform-aws-modules/vpc/aws"
+  version    = "2.15.0"
 
   name               = "${var.cluster_name}-vpc"
   cidr               = var.vpc_cidr
-  azs                = ["${var.region}a"]
+  azs                = var.availability_zones
   enable_nat_gateway = true
-  enable_vpn_gateway = true
+  enable_vpn_gateway = false
 
   enable_dns_hostnames = true
   enable_dns_support   = true
-  public_subnets       = [cidrsubnet(var.vpc_cidr, 4, 0)]
+  public_subnets       = [for k, v in var.availability_zones : cidrsubnet(var.vpc_cidr, 4, k)]
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "owned"
+    "kubernetes.io/role/elb"                    = "1"
+    "cluster.dev/subnet_type"                   = "public"
+  }
+  private_subnets = [for k, v in var.availability_zones : cidrsubnet(var.vpc_cidr, 4, k + 4)]
+  private_subnet_tags = {
+    "kubernetes.io/role/internal-elb" = "1"
+    "cluster.dev/subnet_type"         = "private"
+  }
```

Comment:
Please, add comments about what happens here. Especially in `cidrsubnet()`

Reply:
Done